### ISSUE TITLE
[c++] `SOMAArray`/`ManagedQuery` refactor

### ISSUE THREAD

**Issue by johnkerl** [single-cell-data/TileDB-SOMA#3472]
#3450 is a part, but, the end goal is that we completely remove `ManagedQuery` from `SOMAArray`.

To get there, we need to use `ManagedQuery::read_next` in `Rcpp` instead of `SOMAArray::read_next`. 

[[sc-63693]](https://app.shortcut.com/tiledb-inc/story/63693/somaarray-managedquery-refactor)

**Comment by johnkerl** [comment]
@nguyenv is this completed?

**Comment by nguyenv** [comment]
No, only the read path has been refactored so far. We need to also refactor the write path.
